### ISSUE TITLE
best_model.pth

### ISSUE THREAD

**Issue by Tengau** [VITA-Group/CADTransformer#27]
`=>Failed: no checkpoint found at 'logs//best_model.pth'
`

Hi, I can't seem to find the best_model.pth. Is this released online somewhere? I get this error when I try to run the test code: `CUDA_VISIBLE_DEVICES=0 python -m torch.distributed.launch --nproc_per_node=1 train_cad_ddp.py --data_root /PATH/TO/PROJECT_DIR/data/FloorPlanCAD --pretrained_model /PATH/TO/PROJECT_DIR/pretrained_models/hrnetv2_w48_imagenet_pretrained.pth --test_only`

**Comment by kkkkkom** [comment]
@Tengau , I don't think it is released. But I found you can just do a quick train from scratch, even after epoch 1, the result for some categories are already pretty good.

**Comment by Tengau** [comment]
When I train the model from scratch, I get something like this in the logs, however,  I am a little confused by the imbalance in classes for the GT. Is this the same kind of result you got?

```Epoch 186 (186/300)
2025-02-27 10:29:50,739 Learning rate: 1e-05
2025-02-27 10:29:50,739 BN momentum updated to: 0.01
2025-02-27 10:29:51,460 Train loss: 0.04503, loss seg: 0.04503)
2025-02-27 10:30:44,270 Train loss: 0.04336, loss seg: 0.04336)
2025-02-27 10:31:40,533 Train loss: 0.06607, loss seg: 0.06607)
2025-02-27 10:32:37,192 Train loss: 2e-05, loss seg: 2e-05)
2025-02-27 10:33:00,428 Save last model...
2025-02-27 10:33:13,569 > do validation
2025-02-27 10:33:13,570 > Conducting do_eval
2025-02-27 10:33:16,719 ID:[0  ], CLASS:[BG                  ], Pred Num: [02270.0], GT Num: [02274.0], F1:[87.19%], Precision:[87.27%], Recall:[87.12%]
2025-02-27 10:33:16,720 ID:[1  ], CLASS:[single door         ], Pred Num: [00004.0], GT Num: [00011.0], F1:[26.66%], Precision:[50.00%], Recall:[18.18%]
2025-02-27 10:33:16,720 ID:[2  ], CLASS:[double door         ], Pred Num: [00023.0], GT Num: [00026.0], F1:[69.38%], Precision:[73.91%], Recall:[65.38%]
2025-02-27 10:33:16,720 ID:[3  ], CLASS:[sliding door        ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,720 ID:[4  ], CLASS:[folding door        ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,720 ID:[5  ], CLASS:[revolving door      ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,720 ID:[6  ], CLASS:[rolling door        ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,721 ID:[7  ], CLASS:[window              ], Pred Num: [00039.0], GT Num: [00032.0], F1:[70.42%], Precision:[64.10%], Recall:[78.12%]
2025-02-27 10:33:16,721 ID:[8  ], CLASS:[bay window          ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,721 ID:[9  ], CLASS:[blind window        ], Pred Num: [00004.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,721 ID:[10 ], CLASS:[opening symbol      ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,721 ID:[11 ], CLASS:[sofa                ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,722 ID:[12 ], CLASS:[bed                 ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,722 ID:[13 ], CLASS:[chair               ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,722 ID:[14 ], CLASS:[table               ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,722 ID:[15 ], CLASS:[TV cabinet          ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,722 ID:[16 ], CLASS:[Wardrobe            ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,722 ID:[17 ], CLASS:[cabinet             ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,723 ID:[18 ], CLASS:[gas stove           ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,723 ID:[19 ], CLASS:[sink                ], Pred Num: [00020.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,723 ID:[20 ], CLASS:[refrigerator        ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,723 ID:[21 ], CLASS:[airconditioner      ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,723 ID:[22 ], CLASS:[bath                ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,724 ID:[23 ], CLASS:[bath tub            ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,724 ID:[24 ], CLASS:[washing machine     ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,724 ID:[25 ], CLASS:[squat toilet        ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,724 ID:[26 ], CLASS:[urinal              ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,724 ID:[27 ], CLASS:[toilet              ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,724 ID:[28 ], CLASS:[stairs              ], Pred Num: [00049.0], GT Num: [00079.0], F1:[35.93%], Precision:[46.94%], Recall:[29.11%]
2025-02-27 10:33:16,725 ID:[29 ], CLASS:[elevator            ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,725 ID:[30 ], CLASS:[escalator           ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,725 ID:[31 ], CLASS:[row chairs          ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
2025-02-27 10:33:16,725 ID:[32 ], CLASS:[parking spot        ], Pred Num: [00299.0], GT Num: [00349.0], F1:[82.09%], Precision:[88.96%], Recall:[76.22%]
2025-02-27 10:33:16,725 ID:[33 ], CLASS:[wall                ], Pred Num: [00436.0], GT Num: [00372.0], F1:[55.19%], Precision:[51.15%], Recall:[59.95%]
2025-02-27 10:33:16,725 ID:[34 ], CLASS:[curtain wall        ], Pred Num: [00043.0], GT Num: [00067.0], F1:[30.90%], Precision:[39.53%], Recall:[25.37%]
2025-02-27 10:33:16,726 Total FG Precision:0.6056485176086426, Recall:0.6081932783126831, F1:0.6068682670593262
```

**Comment by kkkkkom** [comment]
@Tengau , hmm, no, I got something like below after epoch 1 (Note: in this run, `nns` was removed from model, so original results from epoch 1 should be even better):

```
Epoch 1 (1/5)
Learning rate: 1e-05
BN momentum updated to: 0.1

100%|██████████████| 6775/6775 [1:49:35<00:00,  1.03it/s, l_seg=0.85, loss=0.85]
Save last model...
> do validation
> Conducting do_eval
100%|█████████████████████████████████████████| 794/794 [03:14<00:00,  4.09it/s]
ID:[0  ], CLASS:[BG                  ], Pred Num: [385060.0], GT Num: [312167.0], F1:[78.80%], Precision:[71.35%], Recall:[88.01%]
ID:[1  ], CLASS:[single door         ], Pred Num: [10988.0], GT Num: [12620.0], F1:[49.10%], Precision:[52.75%], Recall:[45.93%]
ID:[2  ], CLASS:[double door         ], Pred Num: [17341.0], GT Num: [12357.0], F1:[68.56%], Precision:[58.71%], Recall:[82.39%]
ID:[3  ], CLASS:[sliding door        ], Pred Num: [08147.0], GT Num: [05552.0], F1:[44.71%], Precision:[37.60%], Recall:[55.17%]
ID:[4  ], CLASS:[folding door        ], Pred Num: [00000.0], GT Num: [00003.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
ID:[5  ], CLASS:[revolving door      ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
ID:[6  ], CLASS:[rolling door        ], Pred Num: [00000.0], GT Num: [00000.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
ID:[7  ], CLASS:[window              ], Pred Num: [05173.0], GT Num: [10662.0], F1:[37.20%], Precision:[56.95%], Recall:[27.63%]
ID:[8  ], CLASS:[bay window          ], Pred Num: [00000.0], GT Num: [00505.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
ID:[9  ], CLASS:[blind window        ], Pred Num: [02316.0], GT Num: [03092.0], F1:[41.49%], Precision:[48.45%], Recall:[36.29%]
ID:[10 ], CLASS:[opening symbol      ], Pred Num: [00000.0], GT Num: [00921.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
ID:[11 ], CLASS:[sofa                ], Pred Num: [00401.0], GT Num: [04719.0], F1:[7.38%], Precision:[47.13%], Recall:[4.01%]
ID:[12 ], CLASS:[bed                 ], Pred Num: [40747.0], GT Num: [46591.0], F1:[84.65%], Precision:[90.73%], Recall:[79.35%]
ID:[13 ], CLASS:[chair               ], Pred Num: [02623.0], GT Num: [08498.0], F1:[21.58%], Precision:[45.75%], Recall:[14.12%]
ID:[14 ], CLASS:[table               ], Pred Num: [00000.0], GT Num: [05436.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
ID:[15 ], CLASS:[TV cabinet          ], Pred Num: [00000.0], GT Num: [02232.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
ID:[16 ], CLASS:[Wardrobe            ], Pred Num: [16602.0], GT Num: [16380.0], F1:[91.87%], Precision:[91.26%], Recall:[92.50%]
ID:[17 ], CLASS:[cabinet             ], Pred Num: [00922.0], GT Num: [05970.0], F1:[12.39%], Precision:[46.31%], Recall:[7.15%]
ID:[18 ], CLASS:[gas stove           ], Pred Num: [05212.0], GT Num: [07833.0], F1:[68.94%], Precision:[86.28%], Recall:[57.41%]
ID:[19 ], CLASS:[sink                ], Pred Num: [23083.0], GT Num: [24270.0], F1:[78.34%], Precision:[80.36%], Recall:[76.43%]
ID:[20 ], CLASS:[refrigerator        ], Pred Num: [00000.0], GT Num: [02298.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
ID:[21 ], CLASS:[airconditioner      ], Pred Num: [00000.0], GT Num: [02734.0], F1:[0.00%], Precision:[0.00%], Recall:[0.00%]
ID:[22 ], CLASS:[bath                ], Pred Num: [04640.0], GT Num: [06720.0], F1:[44.10%], Precision:[53.99%], Recall:[37.28%]
ID:[23 ], CLASS:[bath tub            ], Pred Num: [01409.0], GT Num: [01310.0], F1:[45.89%], Precision:[44.29%], Recall:[47.63%]
ID:[24 ], CLASS:[washing machine     ], Pred Num: [00059.0], GT Num: [05055.0], F1:[1.64%], Precision:[71.19%], Recall:[0.83%]
ID:[25 ], CLASS:[squat toilet        ], Pred Num: [05344.0], GT Num: [05841.0], F1:[78.30%], Precision:[81.94%], Recall:[74.97%]
ID:[26 ], CLASS:[urinal              ], Pred Num: [01433.0], GT Num: [03547.0], F1:[53.49%], Precision:[92.95%], Recall:[37.55%]
ID:[27 ], CLASS:[toilet              ], Pred Num: [16543.0], GT Num: [15378.0], F1:[80.96%], Precision:[78.11%], Recall:[84.03%]
ID:[28 ], CLASS:[stairs              ], Pred Num: [18152.0], GT Num: [16126.0], F1:[75.16%], Precision:[70.97%], Recall:[79.89%]
ID:[29 ], CLASS:[elevator            ], Pred Num: [12097.0], GT Num: [07374.0], F1:[67.46%], Precision:[54.29%], Recall:[89.07%]
ID:[30 ], CLASS:[escalator           ], Pred Num: [01861.0], GT Num: [01306.0], F1:[27.40%], Precision:[23.32%], Recall:[33.23%]
ID:[31 ], CLASS:[row chairs          ], Pred Num: [22553.0], GT Num: [30563.0], F1:[67.10%], Precision:[79.02%], Recall:[58.31%]
ID:[32 ], CLASS:[parking spot        ], Pred Num: [38017.0], GT Num: [23791.0], F1:[71.94%], Precision:[58.49%], Recall:[93.46%]
ID:[33 ], CLASS:[wall                ], Pred Num: [66894.0], GT Num: [98150.0], F1:[56.74%], Precision:[70.01%], Recall:[47.71%]
ID:[34 ], CLASS:[curtain wall        ], Pred Num: [00688.0], GT Num: [05910.0], F1:[4.76%], Precision:[22.82%], Recall:[2.66%]
Total FG Precision:0.7078456282615662, Recall:0.577660322189331, F1:0.6361114382743835
Save model... Best F1:0.6361114382743835, Best Epoch:0
=> logs/
```

How long did it take for your epoch 1 training? Did the training actually happen?

**Comment by Tengau** [comment]
The loss does seem to go down for each epoch: 
```
Epoch 1 (1/300)
Learning rate: 1e-05
BN momentum updated to: 0.1
Train loss: 3.5335, loss seg: 3.5335)
Train loss: 2.1026, loss seg: 2.1026)
Train loss: 0.49118, loss seg: 0.49118)
Train loss: 0.55253, loss seg: 0.55253)
Save last model...
```

I wonder if the data got corrupted during the preprocessing. When I ran png2svg it took a very long time (3+ hours) for each command. Did you experience something similar?

**Comment by kkkkkom** [comment]
I was initially running preprocessing on kaggle, and it did take a long time. So I moved the preprocessing part to my local machine, which seems to run much faster (couldn't remember the exact time, maybe one hour?).

**Comment by kkkkkom** [comment]
@Tengau , one thing to check is that for hrnet, which model were you using? 
When you load the model weights from `hrnetv2_w48_imagenet_pretrained.pth`, did it complain many missing keys?

**Comment by Tengau** [comment]
Hmm I didn't have an issue like that with the pretrained model. Just to confirm, it should be HRNet-W48-C, and not HRNet-W48-C (w/ CosineLR + CutMix + 300epochs) or HRNet-W48-C-ssld (converted from PaddlePaddle) right?

**Comment by kkkkkom** [comment]
@Tengau, I think I might have misspoken earlier. I looked again, and I think the HRNet-W48-C model is already provided natively in the repository. So, there’s no need to download the model itself—just the model weights for HRNet-W48-C.

I believe my actual question should be: which ViT model did you use? In my case, I had to modify the code to point `default_cfg.default.hf_hub_id="timm/vit_small_patch32_384.augreg_in21k_ft_in1k"` within the `def _create_vision_transformer` function.

Not sure if this is the correct way, but it does seem to work for me and produces reasonable training results.
